include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -783,6 +789,22 @@ class ContiguousInternalMemoryDataFacadeBase : public BaseDataFacade
         return edge_based_node_data.GetTravelMode(id);
     }
 
+    extractor::ClassData GetClassData(const NodeID id) const override final
+    {
+        return edge_based_node_data.GetClassData(id);
+    }
+
+    std::vector<std::string> GetClasses(const extractor::ClassData class_data) const override final
+    {
+        auto indexes = extractor::getClassIndexes(class_data);
+        std::vector<std::string> classes(indexes.size());
+        std::transform(indexes.begin(), indexes.end(), classes.begin(), [this](const auto index) {
+            return m_profile_properties->GetClassName(index);
+        });
+
+        return classes;
+    }
```

Comment:
Hrm the problem is that immediately store these in the `RouteStep` object, which should already own all the contained data, since we might want to expose this object outside of OSRM in the future. (like we already do for all the string data)

 I think I'm going to pass on this optimization since it won't actually help right now. :-/